iothub/service/src/FileUpload/FileUploadNotificationProcessorClient.cs

```diff
@@ -3,40 +3,62 @@
 
 using System;
 using System.IO;
-using System.Net.Mail;
 using System.Net.Sockets;
 using System.Net.WebSockets;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Azure.Amqp;
+using Microsoft.Azure.Devices.Amqp;
 using Microsoft.Azure.Devices.Common;
 using Microsoft.Azure.Devices.Common.Exceptions;
-using Microsoft.Azure.Devices.Common.Extensions;
 
 namespace Microsoft.Azure.Devices
 {
     /// <summary>
     /// Subclient of <see cref="IotHubServiceClient"/> for receiving file upload notifications.
     /// </summary>
     /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-file-upload#service-file-upload-notifications"/>.
-    public class FileUploadNotificationProcessorClient : IDisposable
+    public class FileUploadNotificationProcessorClient
     {
         private readonly string _hostName;
         private readonly IotHubConnectionProperties _credentialProvider;
-        private readonly IotHubConnection _connection;
-        private readonly AmqpFileNotificationReceiver _fileNotificationReceiver;
+        private readonly AmqpConnectionHandler _amqpConnection;
 
         /// <summary>
         /// The callback to be executed each time file upload notification is received from the service.
         /// </summary>
         /// <remarks>
         /// May not be null.
         /// </remarks>
+        /// <example>
+        /// serviceClient.FileUploadNotificationProcessor.FileUploadNotificationProcessor = OnFileUploadNotificationReceived;
+        /// serviceClient.FileUploadNotificationProcessor.OpenAsync();
+        ///
+        /// //...
+        ///
+        /// public AcknowledgementType OnFileUploadNotificationReceived(FileUploadNotification fileUploadNotification)
+        /// {
+        ///    Console.WriteLine($"Received file upload notification from device {fileUploadNotification.DeviceId}")
+        ///    return AcknowledgementType.Complete;
+        /// }
+        /// </example>
         public Func<FileUploadNotification, AcknowledgementType> FileUploadNotificationProcessor;
 
         /// <summary>
         /// The callback to be executed when the connection is lost.
         /// </summary>
+        /// <example>
+        /// serviceClient.FileUploadNotificationProcessor.ErrorProcessor = OnConnectionLost;
+        /// serviceClient.FileUploadNotificationProcessor.OpenAsync();
+        ///
+        /// //...
+        ///
+        /// public void OnConnectionLost(ErrorContext errorContext)
+        /// {
+        ///    // Add reconnection logic as needed
```

Comment:
What would reconnection logic look like? Reinitialize the client? Call open again?

Reply:
I'll link to the sample for this once we've written the sample